test/concurrent/WaitLockTest.ooc

```diff
@@ -0,0 +1,111 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+use concurrent
+use unit
+
+WaitLockTest: class extends Fixture {
+	init: func {
+		super("WaitLock")
+		this add("_testWithMutexOwnership", This _testWithMutexOwnership)
+		this add("_testWithoutMutexOwnership", This _testWithoutMutexOwnership)
+		this add("_testWakeWithFailingCondition", This _testWakeWithFailingCondition)
+		this add("_testWakeWithPassingCondition", This _testWakeWithPassingCondition)
+	}
+	_testWithMutexOwnership: static func {
+		waitLock := WaitLock new()
+		waitLock free()
+	}
+	_testWithoutMutexOwnership: static func {
+		mutex := Mutex new()
+		waitLock := WaitLock new(mutex)
+		waitLock free()
+		mutex free()
+	}
+	_testWakeWithFailingCondition: static func {
+		timesTriggered := 0
```

Comment:
I tried adding `timesTriggered: static Int = 0` to the top of the class and then changing all `timesTriggeredRef@` to `This timesTriggered` and it worked. Please do that.